test/release/examples/primers/learnChapelInYMinutes.chpl

```diff
@@ -850,19 +861,24 @@ class GenericClass {
   var classDomain: domain(1);
   var classArray: [classDomain] classType;
 
-// Explicit constructor.
-  proc GenericClass(type classType, elements : int) {
-    this.classDomain = {1..#elements};
+// Explicit initializer.
+  proc init(type classType, elements : int) {
+    this.classType = classType;
+    this.classDomain = {1..elements};
+    // all generic and const fields must be initialized in "phase 1" prior
+    // to a call to the superclass initializer.
+    super.init(); 
   }
 
-// Copy constructor.
-// Note: We still have to put the type as an argument, but we can
-// default to the type of the other object using the query (``?``) operator.
-// Further, we can take advantage of this to allow our copy constructor
-// to copy classes of different types and cast on the fly.
-  proc GenericClass(other : GenericClass(?otherType),
-                     type classType = otherType) {
+// Copy-style initializer.
+// Note: We include a type argument whose default is the type of the first
+// argument.  This lets our initializer copy classes of different
+// types and cast on the fly.
+  proc init(other : GenericClass(?),
+            type classType = other.classType) {
```

Comment:
Whatever a.init(a) resolves to should be considered a copy initializer. I think this would qualify.
*however* this is a class, not a record, and for classes the compiler doesn't ever actually add a`copy initailize`.